### PR TITLE
Snapshot worker metadata

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: bee-protocol/src/worker/tangle.rs [hunk 1]
```diff
@@ -1,49 +1,51 @@
...
         node.spawn::<Self, _, _>(|shutdown| async move {
-            use tokio::time::interval;
-            use std::time::Duration;
             use futures::StreamExt;
+            use std::time::Duration;
+            use tokio::time::interval;
 
             let mut receiver = ShutdownStream::new(shutdown, interval(Duration::from_secs(1)));
 
+            tangle.update_latest_solid_milestone_index(config.index().into());
```
Comment (zesterer) [inline bee-protocol/src/worker/tangle.rs:37]: Should this be done outside the task? If it's done inside the task, dependent workers may end up with an incomplete view of the tangle.

File: bee-snapshot/src/global/file.rs [hunk 2]
```diff
@@ -37,11 +36,11 @@ pub enum Error {
 }
 
 impl GlobalSnapshot {
-    pub fn from_file(path: &str, index: MilestoneIndex) -> Result<Self, Error> {
+    pub fn from_file(path: &str, index: u32) -> Result<Self, Error> {
```
Comment (zesterer) [inline bee-snapshot/src/global/file.rs:39]: Is it work putting a `type MilestoneIndex = u32;` in `bee-common`? I feel like a lot of these cross-crate types should be put in common with the crates themselves just defining the implementation.

File: bee-tangle/src/tangle.rs [hunk 3]
```diff
@@ -103,6 +103,8 @@ where
 
     /// Inserts a transaction, and returns a thread-safe reference to it in case it didn't already exist.
     pub async fn insert(&self, hash: Hash, transaction: Tx, metadata: T) -> Option<TxRef> {
+        println!("Insert!");
```
Comment (zesterer) [inline bee-tangle/src/tangle.rs:106]: Oh, that can go now.

**Review by zesterer — APPROVED** [review]

File: bee-protocol/src/worker/tangle.rs (hunk 1 above)
Comment (thibault-martinez) [reply bee-protocol/src/worker/tangle.rs]: Done

File: bee-tangle/src/tangle.rs (hunk 3 above)
Comment (thibault-martinez) [reply bee-tangle/src/tangle.rs:106]: I'll let you remove it :p

File: bee-snapshot/src/global/file.rs (hunk 2 above)
Comment (thibault-martinez) [reply bee-snapshot/src/global/file.rs:39]: Yeah maybe... But that would be another PR, it touches too many things to be done now

**Review by zesterer — APPROVED** [review]

**Pull request merged**